### PR TITLE
Support for Ubuntu 18.04

### PR DESCRIPTION
**Pull request by ddeidda**
Signed-off-by: ddeidda <ddeidda@amazon.com>

*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: files/default/sge_preinstall.sh [hunk 1]
```diff
@@ -0,0 +1,3 @@
+#!/bin/sh
+set -e
+wget -cO - $TARBALL_URL > $TARBALL_NAME
```
Comment (lukeseawalker) [inline files/default/sge_preinstall.sh]: missing retry

File: files/ubuntu-18.04/sge_preinstall.sh [hunk 2]
```diff
@@ -0,0 +1,22 @@
+#!/bin/sh
+set -e
+echo "Downloading and extracting source packages for $dest_package"
+
+orig_package='gridengine_8.1.9+dfsg'
+deb_package='gridengine_8.1.9+dfsg-9build1'
+dest_package="$TARBALL_ROOT_DIR"
+package_url='https://bugs.launchpad.net/ubuntu/+archive/primary/+sourcefiles/gridengine/8.1.9+dfsg-9build1'
```
Comment (lukeseawalker) [inline files/ubuntu-18.04/sge_preinstall.sh]: we should use a regionalized link

File: files/ubuntu-18.04/sge_preinstall.sh [hunk 3]
```diff
@@ -0,0 +1,22 @@
+#!/bin/sh
+set -e
+echo "Downloading and extracting source packages for $dest_package"
+
+orig_package='gridengine_8.1.9+dfsg'
```
Comment (lukeseawalker) [inline files/ubuntu-18.04/sge_preinstall.sh]: is it possible to change the cwd of this script to something temporary and not "#{node['cfncluster']['sources_dir']}"?

File: files/ubuntu-18.04/sge_preinstall.sh [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+deb_package='gridengine_8.1.9+dfsg-9build1'
+dest_package="$TARBALL_ROOT_DIR"
+package_url='https://bugs.launchpad.net/ubuntu/+archive/primary/+sourcefiles/gridengine/8.1.9+dfsg-9build1'
+
+wget $package_url/$orig_package.orig.tar.gz
+wget $package_url/$deb_package.debian.tar.xz
+wget $package_url/$deb_package.dsc
```
Comment (lukeseawalker) [inline files/ubuntu-18.04/sge_preinstall.sh]: is there any dpkg or apt command that would avoid to download three packages directly? e.g. apt source

File: files/default/sge_preinstall.sh (hunk 1 above)
Comment (ddeidda) [reply files/default/sge_preinstall.sh]: Ok and replaced with curl

File: files/ubuntu-18.04/sge_preinstall.sh (hunk 2 above)
Comment (ddeidda) [reply files/ubuntu-18.04/sge_preinstall.sh]: Solved by using apt source

File: files/ubuntu-18.04/sge_preinstall.sh (hunk 3 above)
Comment (ddeidda) [reply files/ubuntu-18.04/sge_preinstall.sh]: OK

File: files/ubuntu-18.04/sge_preinstall.sh (hunk 4 above)
Comment (ddeidda) [reply files/ubuntu-18.04/sge_preinstall.sh]: Solved by using apt source

**Pull request merged**